app/collectors/vpc.scala

```diff
@@ -27,17 +28,46 @@ case class AWSVpcCollector(origin:AmazonOrigin, resource: ResourceType, crawlRat
     .overrideConfiguration(AWS.clientConfig)
     .build
 
-  def getSubnets(vpcId: String) = {
+  def getSubnets(vpcId: String): Iterable[AwsSubnet] = {
     val subnetsRequest = DescribeSubnetsRequest.builder.filters(Filter.builder.name("vpc-id").values(vpcId).build).build
     client.describeSubnetsPaginator(subnetsRequest).subnets.asScala
   }
 
+  def getSubnetScopes(vpcId: String): Map[String, SubnetScope] = {
+    val req = DescribeRouteTablesRequest.builder().filters(Filter.builder().name("vpc-id").values(vpcId).build).build
+    val tables = client.describeRouteTablesPaginator(req).routeTables().asScala
+
+    val m = tables.toList.flatMap(table => {
+      // Associations link a route table to other resources, such as subnets,
+      // internet gateways, or nats.
+      val assocs = table.associations().asScala
+
+      val subnetInfo = for {
+        // Discard route tables that do not have an explicit subnet association.
+        subnetAssoc <- assocs.find(a => a.subnetId().nonEmpty)
+
+        // Public subnets have route tables with an associated internet gateway.
+        gatewayAssoc <- assocs.find(a => a.gatewayId().nonEmpty)
+        scope = if (gatewayAssoc.gatewayId().startsWith("igw")) Public else Private
```

Comment:
Worth adding an inline comment here? `startsWith("igw")` can't be the recommended way to detect this?! 😅 🤯

Reply:
Yeah it's a strange one. Have added a comment to explain. Feel like there must be a cleaner way too but can't find it.